hazelcast/src/main/java/com/hazelcast/client/map/impl/querycache/ClientQueryCacheContext.java

```diff
@@ -147,20 +150,29 @@ public Address getThisNodesAddress() {
         throw new UnsupportedOperationException();
     }
 
-    public void recreateAllCaches() {
+    public void recreateAllCaches() throws ExecutionException, InterruptedException {
+        CompletionService<Void> completionService = new ExecutorCompletionService<>(client.getTaskScheduler());
```

Comment:
If I were to choose, I would get rid of the parallelization in here and in ClientCacheProxyFactory. I feel like they are rarely used and the benefits will be small anyways. Also, it makes reasoning about the flow much easier